posthog/api/organization_invite.py

```diff
@@ -37,6 +39,7 @@ class Meta:
             "created_at",
             "updated_at",
             "message",
+            "private_project_access",
```

Comment:
Just as an idea: To validate fields like this where primary keys (=references to other Django models) are inside, you can leverage more of DRF by letting nested serializer do the work. It'll do the lookups and give you objects to work with right away. The specific validation of access is obviously still needed.

```python
class PrivateProjectAccessSerializer(serializers.Serializer):
    id = serializers.PrimaryKeyRelatedField(queryset=Team.objects.all())
    level = serializers.ChoiceField(choices=ExplicitTeamMembership.Level.choices)

    def validate(self, data):
       ...

# and then use it like
    private_project_access = PrivateProjectAccessSerializer(many=True, required=False)

```

Reply:
I struggled with doing this and I'm not 100% certain why - it seemed like it didn't like being given an ID, transformed into an object, and then transformed back into an ID for saving - when I went to get serializer.data it would error. I'm probably missing something simple. However, because I need to get this out for Airbus, and the current method does work, I'm going to merge this in and maybe you can show me how to do what you are suggesting in another PR?

Reply:
I struggled with doing this and I'm not 100% certain why - it seemed like it didn't like being given an ID, transformed into an object, and then transformed back into an ID for saving - when I went to get serializer.data it would error. I'm probably missing something simple. However, because I need to get this out for a customer, and the current method does work, I'm going to merge this in and maybe you can show me how to do what you are suggesting in another PR?